src/inspector_agent.cc

```diff
@@ -648,6 +659,101 @@ void Agent::AllAsyncTasksCanceled() {
   client_->AllAsyncTasksCanceled();
 }
 
+void Agent::InterceptAsyncStackDepthMessage(const StringView& message) {
+  // This logic would be better implemented in JavaScript, but when using
+  // --inspect-brk, the debugger must necessarily attach before much JavaScript
+  // can execute. The Debugger.setAsyncCallStackDepth message arrives too early
+  // and we must intercept this in C++.
+
+  v8::Isolate* isolate = parent_env_->isolate();
+  Local<Context> context = parent_env_->context();
+
+  MaybeLocal<String> string =
+    String::NewFromTwoByte(isolate, message.characters16(),
+                           v8::NewStringType::kNormal, message.length());
+  if (string.IsEmpty()) {
+    return;
+  }
+
+  // Basically, the logic we want to implement is:
+  // let parsed; try {
+  //   parsed = JSON.parse(string);
+  // } catch (e) { return; }
```

Comment:
Is there a `TryCatch` that corresponds to this conceptual `try/catch`? If so, it’s not obvious where… or, rather: If there is none, what happens with the exception if `JSON.parse` throws?

Reply:
I don't believe that the `v8::JSON::Parse` API throws on a parse error (unlike the JavaScript `JSON.parse`). This is the comment I am basing this on: https://cs.chromium.org/chromium/src/v8/include/v8.h?type=cs&l=1815.

/cc @nodejs/v8 if my understanding here is incorrect.

Reply:
Okay, tried it out myself. `v8::JSON::Parse` does indeed throw. I will add a `TryCatch`.

Reply:
@ofrobots Fwiw, as I understand it it’s a general principle of the V8 API that if a `Maybe` method returns an empty value, there is a corresponding pending exception

Reply:
Done.